Genome/Source/BackingDataType.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  BackingDataType.swift
+//  Genome
+//
+//  Created by Logan Wright on 2/10/16.
+//  Copyright © 2016 lowriDevs. All rights reserved.
+//
+
+import Foundation
+
+extension Node: BackingDataType {
+    public init(_ node: Node) {
+        self = node
+    }
+}
+
+public protocol BackingDataType {
+    var isNull: Bool { get }
+    var boolValue: Bool? { get }
+    var doubleValue: Double? { get }
+    var stringValue: String? { get }
+    var arrayValue: [Self]? { get }
+    var objectValue: [String : Self]? { get }
+    
+    init(_ node: Node)
+}
+
+extension BackingDataType {
+    public var floatValue: Float? {
```

Comment:
These should have default implementations as is, but declare in protocol for optional overriding by other data types.

Perhaps `Node` type should declare these types more explicitly as cases to better support various data types.  Not sure though